packages/payload/src/utilities/createLocalReq.ts

```diff
@@ -0,0 +1,63 @@
+import type { Payload, RequestContext } from '..'
+import type { File, PayloadRequest } from '../exports/types'
+
+import { getDataLoader } from '../collections/dataloader'
+import { getLocalI18n } from '../translations/getLocalI18n'
+import getFileByPath from '../uploads/getFileByPath'
+
+function getRequestContext(
+  req: PayloadRequest = { context: null } as PayloadRequest,
+  context: RequestContext = {},
+): RequestContext {
+  if (req.context) {
+    if (Object.keys(req.context).length === 0 && req.context.constructor === Object) {
+      // if req.context is `{}` avoid unnecessary spread
+      return context
+    } else {
+      return { ...req.context, ...context }
+    }
+  } else {
+    return context
+  }
+}
+
+type CreateLocalReq = (
+  options: {
+    collection?: number | string | symbol
+    context?: RequestContext
+    fallbackLocale?: string
+    file?: File
+    filePath?: string
+    locale?: string
+    req?: PayloadRequest
+    user?: Document
+  },
+  payload: Payload,
+) => Promise<PayloadRequest>
+export const createLocalReq: CreateLocalReq = async (
+  { collection, context, fallbackLocale, file, filePath, locale, req = {} as PayloadRequest, user },
+  payload,
+) => {
+  const i18n = req?.i18n || getLocalI18n({ config: payload.config })
+
+  if (payload?.config?.localization) {
+    const defaultLocale = payload?.config?.localization.defaultLocale
+    req.locale = locale || req?.locale || defaultLocale
+    req.fallbackLocale = fallbackLocale || req?.fallbackLocale || defaultLocale
+  } else {
+    req.locale = null
+    req.fallbackLocale = null
+  }
+
+  req.context = getRequestContext(req, context)
+  req.payloadAPI = req?.payloadAPI || 'local'
+  req.payload = payload
+  req.i18n = i18n
+  req.t = i18n.t
+  req.file = file ?? (await getFileByPath(filePath))
```

Comment:
I would consider moving the `getFileByPath` out of this function and into just the operations that need them. That way we can remove the `await` and `async`.

We only need `files` for `create` and `update` right?